@navikt/core/react/src/form/combobox/combobox.stories.tsx

```diff
@@ -0,0 +1,319 @@
+  options,
+  shouldAutocomplete: false,
+};
+
+export function ComboboxWithAddNewOptions(props) {
+  const id = useId();
+  return (
+    <div>
+      <Combobox
+        id={id}
+        label="Komboboks (med nye verdier)"
+        options={props.options}
+        allowNewValues={props.allowNewValues}
+      />
+    </div>
+  );
+}
+
+ComboboxWithAddNewOptions.args = {
+  allowNewValues: true,
+  singleSelect: true,
+  options,
+  shouldAutocomplete: false,
+};
+
+export function ComboboxWithNoHits(props) {
+  const id = useId();
+  const [value, setValue] = useState(props.value);
+  return (
+    <div>
+      <Combobox
+        id={id}
+        label="Komboboks (uten søketreff)"
+        options={props.options}
+        value={value}
+        onChange={(event) => setValue(event.currentTarget.value)}
+      />
+    </div>
+  );
+}
+
+ComboboxWithNoHits.args = {
+  singleSelect: true,
+  options,
+  value: "Orange",
+};
+
+export const Controlled = (props) => {
+  const id = useId();
+  const [value, setValue] = useState(props.value);
+  const [selectedOptions, setSelectedOptions] = useState(props.selectedOptions);
+
+  const onToggleSelected = (option, isSelected) => {
+    if (isSelected) {
+      setSelectedOptions([...selectedOptions, option]);
+    } else {
+      setSelectedOptions(selectedOptions.filter((o) => o !== option));
+    }
+  };
+
+  return (
+    <>
+      <TextField
+        label="Overstyr value"
+        onChange={(event) => setValue(event.target.value)}
+        value={value}
+      />
+      <br />
+      <Combobox
+        label="Hva er dine favorittfrukter?"
+        id={id}
+        options={props.options}
+        onChange={(event) => setValue(event.target.value)}
+        onToggleSelected={onToggleSelected}
+        selectedOptions={selectedOptions}
+        value={value}
+      />
+    </>
+  );
+};
+
+Controlled.args = {
+  value: "apple",
+  options,
+  selectedOptions: initialSelectedOptions,
+};
+
+export const WithCallbacks = () => {
+  const id = useId();
+  const [lastSelected, setLastSelected] = useState<{
+    option: string;
+    isSelected: boolean;
+  }>();
+  return (
+    <div>
+      {lastSelected && (
+        <p>
+          Sist valgt: {lastSelected.option} (
+          {lastSelected.isSelected ? "valgt" : "ikke valgt"})
+        </p>
+      )}
+      <Combobox
+        label="Hva er dine favorittfrukter?"
+        size="medium"
+        id={id}
+        options={options}
+        onToggleSelected={(option, isSelected) =>
+          setLastSelected({ option, isSelected })
+        }
+      />
+    </div>
+  );
+};
+
+WithCallbacks.args = {
+  options: [],
+};
+
+function sleep(ms: number) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+export const CancelInputTest = {
+  render: () => {
+    return (
+      <div data-theme="light">
+        <Combobox options={options} label="Hva er dine favorittfrukter?" />
+      </div>
+    );
+  },
+  play: async ({ canvasElement }) => {
```

Comment:
![image](https://github.com/navikt/aksel/assets/819074/62bad0fc-05d8-4a85-ac6d-df579f53d22c)

Denne skal vel ikke feile? (skriver 'apple' og finner ikke banana?)

Reply:
Skal feile siden en keydown "esc" bør resette listen slik at neste gang man trykker "arrowDown" så vil alle valgene vises. banana er da først i listen med alle valg. Men kan hende et dette er tiltenkt interaksjon 🤔

Reply:
hmm, må ha vært en kode endring kanskje? ser ikke ut til å virke her for meg (FF on macos), den blir tom (esc clearer input, men listen er fortsatt filtrert... very mystisk!)

Reply:
Bra gotcha! Ser ut til å være knyttet til innføringen av "searchTerm"-konseptet. Dette er en slags speiling av "value" som vi innførte med single select-koden. Den oppdateres bare når brukeren skriver i input-feltet, og ikke når teksten autocompletes. Dermed har vi brukt både setValue og setSearchTerm flere steder. Ser ut som vi glemte det i handleClear-funksjonen.

Hadde sett denne feilen tidligere, men glemte den. Bra å få inn tester på det!

Nå kjører storybook-testene, men ikke RTL-testene. Prøver å debugge hvorfor. Håper det er noe caching-greier, for testene er jo omtrent like... 😆

Reply:
Ser ut som koden for escape er `{Escape}`, ikke `{esc}`. Da kjører RTL-testen også. :)